packages/plugin-commands-installation/test/update/update.ts

```diff
@@ -153,3 +153,41 @@ test('update --no-save should not update package.json and pnpm-lock.yaml', async
     expect(lockfile.packages['/peer-a/1.0.1']).toBeTruthy()
   }
 })
+
+// fix: https://github.com/pnpm/pnpm/issues/4196
+test('update should work normal when set empty string version', async () => {
+  await addDistTag({ package: 'peer-a', version: '1.0.1', distTag: 'latest' })
+  await addDistTag({ package: 'peer-c', version: '2.0.0', distTag: 'latest' })
+  await addDistTag({ package: 'pnpm-foo', version: '2.0.0', distTag: 'latest' })
+
+  const project = prepare({
+    dependencies: {
+      'peer-a': '1.0.0',
+    },
+    devDependencies: {
+      'pnpm-foo': '',
+      'peer-c': '',
```

Comment:
I didn't know this was possible. What does it mean? Is it the same as `*`? Does it work with Yarn/npm?

Reply:
works fine in yarn, this was a bug here,  it will update the `devPendencies`(which pkg's version was a empty string `''`) to `dependencies`, it's a wrong behaviour before this pr fix...

Reply:
this works fine in yarn